### PR TITLE
Adjusted functuinality

### PR DESCRIPTION
**Pull request by nikolay-eremeev**
no more Rus or Eng words, only Nr1 and Nr2

### CODE REVIEW

File: app/src/main/java/com/example/myapplicationjava/Dictionary.java [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
 
     private List<Learner> data;
     private Learner mCurrentLearn;
     private File mFile;
+    private int ind;
 
     Dictionary() {
         data = new ArrayList();
-        mFile = new File("/sdcard/download/dctnr.txt");
+        mFile = new File(Environment.getDownloadCacheDirectory().getPath() + "/dctnr.txt");
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/Dictionary.java]: Все string - в файл для их хранения и использовать по ключу

File: app/src/main/java/com/example/myapplicationjava/Dictionary.java [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
+
 import java.io.File;
 import java.io.FileNotFoundException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Random;
 import java.util.Scanner;
 
-public class Dictionary {
+class Dictionary {
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/Dictionary.java]: Использовать data class

File: app/src/main/java/com/example/myapplicationjava/Dictionary.java [hunk 3]
```diff
@@ -1,21 +1,24 @@
...
 
     private List<Learner> data;
     private Learner mCurrentLearn;
     private File mFile;
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/Dictionary.java]: Почему mFile вынесена как переменная, если ее инициализация происходит ровно 1 раз и больше она нигде в классе не используется? Тот же вопрос и к остальным переменным.

File: app/src/main/java/com/example/myapplicationjava/Dictionary.java [hunk 4]
```diff
@@ -31,28 +34,25 @@
 
     }
 
-    public void pickToLearn() {
+    void pickToLearn() {
         Random rand = new Random();
         String[] p = {"","","","",""};
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/Dictionary.java]: Что за название? Непонятно для чего используется переменная

File: app/src/main/java/com/example/myapplicationjava/Dictionary.java [hunk 5]
```diff
@@ -31,28 +34,25 @@
...
 
     }
 
-    public String getEng() {
-        return mCurrentLearn.getWord1();
-    }
-
-    public String getRus() {
-        return mCurrentLearn.getWord2();
+    String getWord() {
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/Dictionary.java]: Use data class, you can setup all properties and not only get and set. https://kotlinlang.org/docs/reference/data-classes.html

File: app/src/main/java/com/example/myapplicationjava/Dictionary.java [hunk 6]
```diff
@@ -31,28 +34,25 @@
...
+        return mCurrentLearn.getWord(ind);
     }
 
-    public String getTranslation(String toTranslate) {
-        return mCurrentLearn.getTranslation(toTranslate);
+    String getTranslation(String toTranslate) {
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/Dictionary.java]: Use correct naming for passing variable - I mean `String getTranslation(String word)` - it's enough, cuz toTranslate - doesn't show what do you want. Or  `String getTranslationFor(String word)`

File: app/src/main/java/com/example/myapplicationjava/Learner.java [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 public class Learner {
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/Learner.java]: Why do you need this class? Use it in your data class or interface with default realization. And your data class - will inheritance it.

File: app/src/main/java/com/example/myapplicationjava/Learner.java [hunk 8]
```diff
@@ -24,23 +24,16 @@ public double getRate() {
...
-        return mWord1;
-    }
-
-    public String getWord2() {
-        return mWord2;
-    }
-
-    public String getTranslation(String word) {
-        if (word.equals(mWord2))
+    String getWord(int nr) {
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/Learner.java]: Naming...

File: app/src/main/java/com/example/myapplicationjava/Learner.java [hunk 9]
```diff
@@ -24,23 +24,16 @@ public double getRate() {
...
+        if (nr == 0)
             return mWord1;
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/Learner.java]: Why do you use "m" prefix?

File: app/src/main/java/com/example/myapplicationjava/MainActivity.java [hunk 10]
```diff
@@ -63,11 +75,27 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
             }
         });
 
+        mButtonNext.setOnClickListener(new View.OnClickListener() {
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/MainActivity.java]: Use the separate method for handle onclickListener. Not into `onCreate`. And move into it all operations with handling.

File: app/src/main/java/com/example/myapplicationjava/MainActivity.java [hunk 11]
```diff
@@ -63,11 +75,27 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
...
+            public void onClick(View v) {
+                mTextViewRemember.setText(
+                        mDictionary.getWord() + "\n" +
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/MainActivity.java]: Combine in one function.

File: app/src/main/java/com/example/myapplicationjava/MainActivity.java [hunk 12]
```diff
@@ -1,23 +1,29 @@
...
 import android.widget.EditText;
 import android.widget.TextView;
 
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+
 public class MainActivity extends AppCompatActivity {
     
     private TextView mTextViewLearnWord;
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/MainActivity.java]: You have an id for all elements in your layout. You should use it. Don't need a lot of code unnecessary code.

Comment (mirragato) [reply app/src/main/java/com/example/myapplicationjava/MainActivity.java]: https://kotlinlang.org/docs/tutorials/android-plugin.html

File: app/src/main/res/layout/activity_main.xml [hunk 13]
```diff
@@ -12,39 +12,39 @@
         android:layout_height="wrap_content"
         android:layout_marginStart="8dp"
         android:layout_marginLeft="8dp"
-        android:layout_marginTop="8dp"
+        android:layout_marginTop="40dp"
```
Comment (mirragato) [inline app/src/main/res/layout/activity_main.xml]: Do you test it for all screens?

File: app/src/main/res/layout/activity_main.xml [hunk 14]
```diff
@@ -12,39 +12,39 @@
...
-        android:layout_marginTop="8dp"
+        android:layout_marginTop="80dp"
         android:layout_marginEnd="8dp"
         android:layout_marginRight="8dp"
-        android:layout_marginBottom="8dp"
+        android:layout_marginBottom="120dp"
         android:ems="10"
         android:gravity="center"
         android:inputType="textPersonName"
-        android:text="Name"
```
Comment (mirragato) [inline app/src/main/res/layout/activity_main.xml]: Move into the special file for a string.

File: app/src/main/res/layout/activity_main.xml [hunk 15]
```diff
@@ -57,10 +57,66 @@
...
+        android:textSize="18sp"
         app:layout_constraintBottom_toTopOf="@+id/textView"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.07999998" />
+
+    <Button
+        android:id="@+id/button2"
```
Comment (mirragato) [inline app/src/main/res/layout/activity_main.xml]: Correct naming for id

File: app/src/main/java/com/example/myapplicationjava/MainActivity.java [hunk 16]
```diff
@@ -34,10 +40,16 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        mTextViewRemember = findViewById(R.id.textView4);
+        mTextViewRemember.setText(mDictionary.getWord());
 
         mTextViewResult = findViewById(R.id.textView2);
 
+        mButtonNext = findViewById(R.id.button2);
+        mButtonClean = findViewById(R.id.button3);
+
         mEditTextUserInput = findViewById(R.id.editText);
         mEditTextUserInput.addTextChangedListener(new TextWatcher() {
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/MainActivity.java]: userText.equals("") - isEmpty
mDictionary.getTranslation(questionedText).equals(userText) - create fanction which returns boolean, true or false - and set text after result.

Use placeholder in textField.
```
mTextViewResult.setText("ENTER THE TRANSLATION");
                    mTextViewResult.setTextColor(Color.BLACK);
```

File: app/src/main/java/com/example/myapplicationjava/Dictionary.java (hunk 2 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/Dictionary.java]: it is JAVA, so no data class structures, I implement it on my own

File: app/src/main/java/com/example/myapplicationjava/Dictionary.java (hunk 5 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/Dictionary.java]: it is JAVA, so no data class structures, I implement it on my own

File: app/src/main/java/com/example/myapplicationjava/Learner.java (hunk 7 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/Learner.java]: it is JAVA, so no data class structures, I implement it on my own

File: app/src/main/java/com/example/myapplicationjava/Learner.java (hunk 9 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/Learner.java]: Private, non-static field names start with m.
https://github.com/ribot/android-guidelines/blob/master/project_and_code_guidelines.md

File: app/src/main/res/layout/activity_main.xml (hunk 13 above)
Comment (nikolay-eremeev) [reply app/src/main/res/layout/activity_main.xml]: no, I will need your assistance

File: app/src/main/java/com/example/myapplicationjava/Dictionary.java (hunk 1 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/Dictionary.java]: done

File: app/src/main/java/com/example/myapplicationjava/Dictionary.java (hunk 3 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/Dictionary.java]: done

File: app/src/main/java/com/example/myapplicationjava/Dictionary.java (hunk 4 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/Dictionary.java]: done

File: app/src/main/java/com/example/myapplicationjava/MainActivity.java (hunk 12 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/MainActivity.java]: done

File: app/src/main/res/layout/activity_main.xml (hunk 14 above)
Comment (nikolay-eremeev) [reply app/src/main/res/layout/activity_main.xml]: done

File: app/src/main/java/com/example/myapplicationjava/MainActivity.java (hunk 16 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/MainActivity.java]: done

File: app/src/main/java/com/example/myapplicationjava/Card.java [hunk 17]
```diff
@@ -13,34 +19,23 @@ public Learner (String [] args) {
...
     }
 
-    public String getWord2() {
-        return mWord2;
-    }
-
-    public String getTranslation(String word) {
-        if (word.equals(mWord2))
+    String getWord(int nr) {
+        if (nr == 0)
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/Card.java]: Use switch - it'll look more pretty

File: app/src/main/java/com/example/myapplicationjava/Card.java [hunk 18]
```diff
@@ -1,8 +1,14 @@
 package com.example.myapplicationjava;
 
-public class Learner {
+public class Card {
 
-    public Learner (String [] args) {
+    private String mLanguage1;
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/Card.java]: Why do you use "m" prefix?

File: app/src/main/java/com/example/myapplicationjava/LearningCards.java [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+
+        if (mData.size() == 0) {
+            //TODO handle the case when mData has no entries
+        }
+        else {
+            int randIndex = rand.nextInt(mData.size());
+            mCurrentCard = mData.get(randIndex);
+            mCurrentWordNr = rand.nextInt(2);
+        }
+
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/LearningCards.java]: See a guideline https://google.github.io/styleguide/javaguide.html and apply it everywhere in code.

File: app/src/main/java/com/example/myapplicationjava/LearningCards.java [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    String getWord() {
+        return mCurrentCard.getWord(mCurrentWordNr);
+    }
+
+    String getTranslation() {
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/LearningCards.java]: For` getWord()` and `getTranslation()` can use one function and pass there, for instance, Boolean. Think about how it'll do better

File: app/src/main/java/com/example/myapplicationjava/LearningCards.java [hunk 21]
```diff
@@ -0,0 +1,66 @@
...
+        return mCurrentCard.getWord(1 - mCurrentWordNr);
+    }
+
+    boolean isTranslationRight(String userText) {
+
+        String userTextTmp = userText.replaceAll("[^a-zA-Z а-яА-Я]", "").toLowerCase();
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/LearningCards.java]: I see copypaste :) Think how to combine it, idk about extensions in java, but you can find another solution

File: app/src/main/java/com/example/myapplicationjava/MainActivity.java [hunk 22]
```diff
@@ -1,44 +1,77 @@
...
+    private LearningCards mLearningCards;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        mDictionary = new Dictionary();
 
-        if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        int MY_PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE = 0;
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/MainActivity.java]: Why does it need like a variable?

File: app/src/main/java/com/example/myapplicationjava/MainActivity.java [hunk 23]
```diff
@@ -48,26 +81,39 @@ public void beforeTextChanged(CharSequence s, int start, int count, int after) {
...
                 String userText = mEditTextUserInput.getText().toString();
-                String questionedText = mTextViewLearnWord.getText().toString();
-                if (userText.equals("")) {
-                    mTextViewResult.setText("ENTER THE TRANSLATION");
-                    mTextViewResult.setTextColor(Color.BLACK);
-                } else if (mDictionary.getTranslation(questionedText).equals(userText)) {
-                    mTextViewResult.setText("GOOD, NEXT WORD IS:");
-                    mTextViewResult.setTextColor(Color.GREEN);
+                if (userText.isEmpty()) {
+                    mTextViewResult.setText(getResources().getString(R.string.empty));
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/MainActivity.java]: Use one function and pass ids

File: app/src/main/java/com/example/myapplicationjava/MainActivity.java (hunk 22 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/MainActivity.java]: removed

File: app/src/main/java/com/example/myapplicationjava/LearningCards.java (hunk 21 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/LearningCards.java]: created

File: app/src/main/java/com/example/myapplicationjava/LearningCards.java (hunk 20 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/LearningCards.java]: names are good

File: app/src/main/java/com/example/myapplicationjava/Card.java (hunk 18 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/Card.java]: removed

File: app/src/main/java/com/example/myapplicationjava/Card.java (hunk 17 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/Card.java]: done

File: app/src/main/java/com/example/myapplicationjava/MainActivity.java (hunk 23 above)
Comment (nikolay-eremeev) [reply app/src/main/java/com/example/myapplicationjava/MainActivity.java]: didn't get what you mean. there is a new version

File: app/src/main/java/com/example/myapplicationjava/LaunchActivity.java [hunk 24]
```diff
@@ -0,0 +1,108 @@
...
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()) {
+
+            case R.id.buttonNext:
+                String userText = etUserInput.getText().toString();
+                etUserInput.getText().clear();
+                tvRemember.setText(
+                        myLearningCards.getWord()
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/LaunchActivity.java]: Combine it inside myLearningCards

File: app/src/main/java/com/example/myapplicationjava/LearningCards.java [hunk 25]
```diff
@@ -0,0 +1,98 @@
...
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Random;
+import java.util.Scanner;
+
+class LearningCards {
+
+    private static final String TAG = "LearningCards";
+    private ArrayList<Card> mData;
```
Comment (mirragato) [inline app/src/main/java/com/example/myapplicationjava/LearningCards.java]: prefix "m"

Comment (mirragato) [reply app/src/main/java/com/example/myapplicationjava/LearningCards.java]: Cards

**Pull request merged**